audiomentations/augmentations/duration_clip.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+from audiomentations.core.transforms_interface import BaseWaveformTransform
+
+
+class DurationClip(BaseWaveformTransform):
+    """
+    Clip audio by duration specified in samples or seconds.
+    If the input sound is longer than the target duration,
+    pick a random offset and crop the sound to the target duration.
+    If the input sound is shorter than the target duration,
+    pad the end of the sound (append digital silence) so the duration matches the target duration.
+    """
+
+    supports_multichannel=True
+
+    def __init__(
+        self,
+        duration_samples: int = None,
+        duration_seconds: float = None,
+        pad_mode: str = "constant",
+        pad_section: str = "end",
+        p: float = 0.5,
+    ):
+        """
+        :param duration_samples: Target duratin in sample length
+        :param duration_seconds: Target duratin in second
+        :param pad_mode: Padding mode. Must be supported by `numpy.pad`
+        :param pad_section: Which part of the signal should be replaced with padding:
+            "start" or "end"
+        :param p: The probability of applying this transform
+        """
+        super().__init__(p)
+        self.mode = pad_mode
+
+        assert pad_section in ("start", "end")
+        self.pad_section = pad_section
+
+        assert duration_samples is not None or duration_seconds is not None
+        if duration_samples is not None and duration_seconds is not None:
+            raise ValueError(
+                "should have duration_samples or duration_seconds, but not both"
+            )
+        elif duration_seconds:
+            assert duration_seconds > 0
+            self.get_target_samples = lambda sr: int(duration_seconds * sr)
+        elif duration_samples:
+            assert duration_samples > 0
+            self.get_target_samples = lambda sr: duration_samples
```

Comment:
Raise an exception if both duration_samples _and_ duration_seconds are given. Only one of them should be given.

Reply:
"both are given" exception is raised in line 40.
"none of them are given" exception is an assertion, and it at line 39.
Now the expression is indeed relatively unclear，I change it in another suggestion.